pkg/deploy-manager/subscription.go

```diff
@@ -320,6 +354,131 @@ func (t *DeployManager) DeployOCSWithOLM(ocsRegistryImage string, localStorageRe
 	return nil
 }
 
+// UpgradeOCSWithOLM upgrades ocs operator via an olm subscription
+func (t *DeployManager) UpgradeOCSWithOLM(ocsRegistryImage string, localStorageRegistryImage string, subscriptionChannel string) error {
+
+	if ocsRegistryImage == "" || localStorageRegistryImage == "" {
+		return fmt.Errorf("catalog registry images not supplied")
+	}
+
+	co := t.generateClusterObjects(ocsRegistryImage, localStorageRegistryImage, subscriptionChannel)
+	err := t.updateClusterObjects(co)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// UninstallOCS uninstalls ocs operator and storage clusters
+func (t *DeployManager) UninstallOCS(ocsRegistryImage string, localStorageRegistryImage string, subscriptionChannel string) error {
```

Comment:
i think we'll end up simplifying this logic soon, but this looks like a good port of the cluster-clean.sh script as it is today.